internal_transfers/actions/src/simulate/interface.ts

```diff
@@ -0,0 +1,31 @@
+export abstract class TxSimulator {
```

Comment:
Not sure why here you would use an abstract class instead of an interface. 

Given the implementation of this class i don't see clear advantages of this.

Reply:
Probably because I don't know the difference. I was just seeking a structure that had unimplemented function signatures. Perhaps you can elaborate on what I should be doing here.

Reply:
I think there's a lot of literature around this. In general favour always interfaces over classes. 

Interfaces gives you the typechecks you seek, but will be removed from code. 
They can't have code associated with them. They are perfect for what you want to achieve here, to define a signature for a method that would do a simulation of a calldata and return some info you care about. You can be explicit with what are the params and return types you seek

Reply:
I definitely looked around for defining function signatures in an interface (before choosing abstract class) and still have not found this...

https://www.typescriptlang.org/docs/handbook/interfaces.html

are these just supposed to be loosely defined outside the scope of the interface itself?